src/saas_shield/deterministic.rs

```diff
@@ -175,6 +161,83 @@ impl DeterministicFieldOps for SaasShieldDeterministicClient {
             .collect()
     }
 
+    async fn rotate_fields(
+        &self,
+        encrypted_fields: EncryptedFields,
+        metadata: &AlloyMetadata,
+        new_tenant_id: TenantId,
+    ) -> Result<RotateBatchResult, AlloyError> {
+        let paths = encrypted_fields
+            .values()
+            .map(|field| (field.secret_path.clone(), field.derivation_path.clone()))
+            .collect_vec();
+        let original_tenant_keys = derive_keys_many_paths(
+            &self.tenant_security_client,
+            metadata,
+            paths.clone(),
+            SecretType::Deterministic,
+        )
+        .await?;
+        let new_tenant_keys = if new_tenant_id != metadata.tenant_id {
+            derive_keys_many_paths(
+                &self.tenant_security_client,
+                metadata,
+                paths,
+                SecretType::Deterministic,
+            )
+            .await?
+        } else {
+            original_tenant_keys.clone() // TODO: I only need a reference to this, but the `else` creates an owned value.
```

Comment:
I don't have a better solution off the top of my head.

Reply:
:+1: